xmtp_mls/src/subscriptions.rs

```diff
@@ -242,7 +243,12 @@ where
             futures::pin_mut!(stream);
             let _ = tx.send(());
             while let Some(convo) = stream.next().await {
-                convo_callback(convo)
+                let provider = client.context.mls_provider()?;
+                // Don't execute callback for dms unless include_dm is true
+                if include_dm || convo.metadata(provider)?.conversation_type != ConversationType::Dm
```

Comment:
it would be better to do this filter in `stream_conversations` itself rather than the callback method -- the callback methods should retain the same behavior as the Non-callback variants, but just accept a function instead. this makes the logic for both easier to follow

Here is a place you could insert a check/filter for DM Group (i.e another filter/filter map call just above that, or even just save the iterator and if the dm_group flag is on, filter the iterator again): https://github.com/xmtp/libxmtp/blob/998623d7d3a9fd1a1ecdc08779a144f6b9c53105/xmtp_mls/src/subscriptions.rs#L160

Reply:
good call 👌 , fixed here fc02806f7e7f47b402bb019c56379112d4fda97e